website/docs/r/slo.html.markdown

```diff
@@ -0,0 +1,153 @@
+---
+layout: 'sumologic' page_title: 'SumoLogic: sumologic_slo' description: |- Provides the ability to create, read, delete,
+and update SLOs.
+---
+
+# sumologic_slo
+
+Provides the ability to create, read, delete, and update [SLOs][1].
+
+## Example SLO
+
+```hcl
+resource "sumologic_slo" "slo_tf_window_based" {
+  name        = "slo-tf-window-based"
+  description = "example SLO created with terraform"
+  parent_id   = "0000000000000001"
+  signal_type = "Latency"
+  service     = "auth"
+  application = "login"
+  compliance {
+    compliance_type = "Rolling"
+    size            = "7d"
+    target          = 99
+    timezone        = "Asia/Kolkata"
+  }
+  indicator {
+    window_based_evaluation {
+      op         = "LessThan"
+      query_type = "Metrics"
+      size       = "1m"
+      threshold  = 200
+      queries {
+        query_group_type = "Threshold"
+        query_group {
+          row_id        = "A"
+          query         = "metric=request_time_p90  service=auth api=login"
+          use_row_count = false
+        }
+      }
+    }
+  }
+}
+
+resource "sumologic_slo" "slo_tf_request_based" {
+  name        = "slo-tf-request-based"
+  description = "example SLO created with terraform for request based SLI"
+  parent_id   = sumologic_slo_folder.tf_slo_folder.id
+  signal_type = "Latency"
+  service     = "auth"
+  application = "login"
+  compliance {
+    compliance_type = "Rolling"
+    size            = "7d"
+    target          = 99
+    timezone        = "Asia/Kolkata"
+  }
+  indicator {
+    request_based_evaluation {
+      op         = "LessThanOrEqual"
+      query_type = "Logs"
+      threshold  = 1
+      queries {
+        query_group_type = "Threshold"
+        query_group {
+          row_id        = "A"
+          query         = <<QUERY
+          cluster=sedemostaging namespace=warp004*
+              | parse "Coffee preparation request time: * ms" as latency nodrop
+              |  if(isBlank(latency), "false", "true") as hasLatency
+              | where hasLatency = "true"
+              |  if(isBlank(latency), 0.0, latency) as latency
+              | latency/ 1000 as latency_sec
+QUERY
+          use_row_count = true
+          field = "latency_sec"
+        }
+      }
+    }
+  }
+}
+```
+
+## Argument reference
+
+The following arguments are supported:
+
+- `name` - (Required) The name of the SLO. The name must be alphanumeric.
+- `description` - (Optional) The description of the SLO.
+- `parent_id` - (Optional) The ID of the SLO Folder that contains this SLO. Defaults to the root folder.
+- `signal_type` - (Required) The type of SLO. Valid values are `Latency`, `Error`, `Throughput`, `Availability`
+  , `Other`. Defaults to `Latency`.
+- `service` - (Optional) The notifications the SLO will send when the respective trigger condition is met.
+- `application` - (Optional) Whether to group notifications for individual items that meet the trigger condition.
```

Comment:
Docs are still incorrect.

Reply:
the `service` and `application ` fields ? me & @tarunk2 discussed this and concluded we should keep it for compatibility with openslo schema. also if the config is stored anyway then it makes no sense for users to wait for it to be functional and change the tf files and reapply it in future

Reply:
the description of these fields is incorrect.